src/java/org/apache/cassandra/service/accord/txn/TxnQuery.java

```diff
@@ -217,6 +217,9 @@ public Result compute(TxnId txnId, Timestamp executeAt, Seekables<?, ?> keys, @N
                 ClientRequestsMetricsHolder.accordWriteMetrics.accordMigrationRejects.mark();
             else
                 ClientRequestsMetricsHolder.accordReadMetrics.accordMigrationRejects.mark();
+            // Prevent writes from being applied
+            if (update != null)
+                ((TxnUpdate)update).failCondition();
```

Comment:
does a recovery coordinator always evaluate this the same way?

Reply:
It should be deterministic. Live migration is predicated on this being true. That said the checks `TxnQuery` still needs to be updated to be done using the information from the actual epoch rather than the arbitrary current epoch.